### ISSUE TITLE
Version 8.0.6 – RequestsTree (sidebar)  enhansments

### ISSUE THREAD

**Issue by kool79** [a85/POSTMan-Chrome-Extension#9659]
Version 8.0.6 
I want to report about issues/ask for few enhansments when working with RequestsTree.

**_Issue#1_**: We have too many tabs with requests. Every time when I click item in requests tree I have +1 opened tab.  But sometimes I want just to select (highlight) item in the tree without openeing it. For example, I do that before I scroll the tree to not loose my context. **Proposed solution:**  select tab with single-click and open it with double-click.

**_Issue#2_**: (related to issue#1) Naviagation with keyboard on RequestsTree: When I use keyboard arrows to move over tree items ALL those items are opened in right window. I.e. every request is opened after it gots focus in tree. Problem: too many windows opened in right pane, difficult to keep working context.  **Proposed solution:** open item only when I press 'Enter'/Return key.  Note: but this seems like a bug because in settings>shortcuts we have a separete actions for up/down arrows (prev/next item) and separate action for 'Enter' (select item)

**_Issue#3_**: 
    When I have request with examples and I click this request in RequestsTree the examples are automatically expanded. It is very annoying to collapse them all again if I want to be focused of requests only. When we have many examples they occupy whole tree and for big collections I need to scroll the RequestsTree.
    Same for folders/collections:  sometimes I just want  to run all requests from folder/collection without expanding it. But when I click folder/collection it is automatically expanded. This is not convinient in a case, when I want to run folder as perequisite (for example, "do_register_and_login" folder) and then I want to call another endpoint which is located in a tree below: When I select folder do_register_and_login it expanded and then I need to scroll again to the target endpoint (or collapse do_register_and_login folder)
     **Proposed solution:** expand items in RequestsTree only when user ">" icon (or press 'right arrow' key on keyboard)

**_Issue#4_**:  Again, too many open tabs. I need an option "Close all but unsaved". Because "Close all" somewtimes leaves unsaved tabs opened if I discard to to save unsaved tab.

**_Issue#5_**:  Again, too many open tabs and no ui whrer I can see a full list of opened tabs (with fill names). I can scroll right and left but In tab titles I see method names and only first 1 or 2 characters from method name:
![Screenshot - 24 02 2021 , 20_59_50](https://user-images.githubusercontent.com/4625317/109051500-47d92900-76e3-11eb-81b5-cfac1c161568.png)
  I know that we have hint with full names but it gets too much time to find required tab:  mouse-over every tab, wait 1 second, do that for each tab, scroll to next portion. 
  **Proposed solution#1:** - separate window with list of active tabs. **Proposed solution#2:** popup with with list of active tabs.  It can be the same as "recently closed" tabs but with another name: ("All opened tabs")
![Screenshot - 24 02 2021 , 21_08_01](https://user-images.githubusercontent.com/4625317/109052552-6855b300-76e4-11eb-82a9-da6b3884f3d4.png)

**Comment by aaginsa** [comment]
Agree with issue#3, auto expand examples when click on request inside collection are really annoying.
For now i downgrade my postman to v7 again

**Comment by ps2goat** [comment]
Issue #6: I can't see the full name of my collection because the sidebar has a limit to how wide it can be. I thought it was due to the scaling on my UI, but `CTRL + -` has no effect on how much I can see in the sidebar. The matter only gets worse when working with forks, as the branch name takes up a sizable chunk of screen space. I liked the old view better where the branch name was under the collection name (or maybe it wrapped?)

**Comment by Quaternions** [comment]
Issue#3 was the reason I came here.  I was going to suggest we have a setting option that allow us to turn off or on examples in the collection tree.  I'm working a major project with 3 backend teams.  Our API currently has over 200 endpoints currently.  These examples make browsing the collections very frustrating.  Especially when the examples auto-expand every time you double click on an endpoint to open it up in a new tab.

Is there any way we can add the examples back to the endpoint tab once it's open?  Having them in the collection tree isn't practical for people who are working with large collections.

Screenshot of one of my collections:
![image](https://user-images.githubusercontent.com/6989440/111500164-12d85900-871a-11eb-804f-c8cf3fd0d398.png)

**Comment by dliwustl** [comment]
Yes, could we **please** disable the auto-expanding behavior in Issue #3? There are already carets there, so let those be used to expand the collection tree.

**Comment by snotmare** [comment]
+1 from me on #3. I would like to see an option that completely turns off the auto focus in the tree when you switch between tabs.

Perhaps you could add a right-click option on the tab to "Reveal in tree" that will focus on that request. This could work similar to VS Code in that way.

Thanks!

**Comment by vieiraes** [comment]
worst bug/feature of ever!

**Comment by rtritean** [comment]
+1 for #3
